internal/controller/nnf_workflow_controller_container_helpers.go

```diff
@@ -420,7 +427,7 @@ func (c *nnfUserContainer) applyPermissions(spec *corev1.PodSpec, mpiJobSpec *mp
 		container := &spec.Containers[idx]
 
 		// Add an InitContainer to map the user to the provided uid/gid using /etc/passwd
-		c.addInitContainerPasswd(spec, container.Image)
+		c.addInitContainerPasswd(spec, container.Image, idx)
```

Comment:
I don't get this. Wouldn't you need just one init container to create the /etc/passwd, and then mount that into all other containers?

Reply:
The way it works now is you get an InitContainer for every container and they cannot have the same names. What you're suggesting is an improvement and I will do that as a follow-up.